core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -565,8 +573,14 @@ private[spark] class ExecutorAllocationManager(
     } else {
       // We don't want to change our target number of executors, because we already did that
       // when the task backlog decreased.
-      client.killExecutors(executorIdsToBeRemoved.toSeq, adjustTargetNumExecutors = false,
-        countFailures = false, force = false)
+      if (conf.get(WORKER_DECOMMISSION_ENABLED)) {
```

Comment:
I see that we are not saving the config value at initialization time and are instead checking the conf on each round -- is that so that you can disable this setting mid flight and revert to killing (instead of decommissioning) without restarting the SparkContext ?

Reply:
Just seemed easier to write, I can move it into a private val if you'd prefer.

Reply:
I think a private val would make it easier to read and also consistent: We know upfront whether we are doing decommissioning or not without the chance of some codepaths doing decommissioning and then some later not (when the config changes).